bin/mariadb

```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use lib '/usr/local/pf/lib';
+
+use pf::services::util;
+use POSIX qw(:signal_h pause _exit setuid setgid);
+use pf::log;
+use pf::cluster;
+use pf::config qw(
+    %Config
+);
+use pf::util;
+use DBI;
+
+our $PROGRAM_NAME = $0 = "pf-mariadb";
+
+my $logger = get_logger( $PROGRAM_NAME );
+
+our $RUNNING = 1;
+
+my (undef, undef,$uid,$gid) = getpwnam('mysql');
+setuid($uid);
+setgid($gid);
+
+our $MYSQLD_PID;
+our $IS_CHILD;
+
+# How many times should we attempt to connect to a galera cluster when an alive quorum is there
+# After this, the management node will declare a new cluster if its not in maintenance mode
+# TODO: move this to configuration
+our $PEERS_ALIVE_CONNECT_ATTEMPT = 1;
+
+# Time to wait between starts after a failure
+# TODO: move this to configuration
+our $WAIT_TIME_FAIL = 10;
+
+# Time to wait between attempts to discover alive servers
+# TODO: move this to configuration
+our $WAIT_TIME_ALIVE = 10;
+
+our $failed_counts = {};
+
+# init signal handlers
+POSIX::sigaction(
+    &POSIX::SIGHUP,
```

Comment:
If we are going to define a sighandler for SIGHUP, might it not do something more useful than killing the process?